pkg/descheduler/strategies/rescheduler_mode.go

```diff
@@ -0,0 +1,199 @@
+		return err
+	}
+	return nil
+}
+
+// releaseTaintsFromNode releases criticalAddonsOnlyTaint from node.
+func releaseTaintsFromNode(client clientset.Interface, node *v1.Node) {
+	// Get the updated node object since, we have applied taint in addTaints function. Without this we'd get
+	// error related to node object being old.
+	node, err := client.CoreV1().Nodes().Get(node.Name, metav1.GetOptions{})
+	newTaints := make([]v1.Taint, 0)
+	for _, taint := range node.Spec.Taints {
+		if taint.Key == criticalAddonsOnlyTaintKey {
+			glog.Infof("Releasing taint %+v on node %v", taint, node.Name)
+		} else {
+			newTaints = append(newTaints, taint)
+		}
+	}
+
+	if err != nil {
+		glog.Fatal("Error while getting updated node object")
+	}
+	if len(newTaints) != len(node.Spec.Taints) {
+		node.Spec.Taints = newTaints
+		_, err := client.CoreV1().Nodes().Update(node)
+		if err != nil {
+			glog.Warningf("Error while releasing taints on node %v: %v", node.Name, err)
+		} else {
+			glog.Infof("Successfully released all taints on node %v", node.Name)
+		}
+	}
+}
+
+// getResourceUsageOfPod get the cpu and memory usage of given pod.
+func getResourceUsageOfPod(pod *v1.Pod) (int64, int64) {
+	var cpuUsed, memoryUsed int64
+	for _, container := range pod.Spec.Containers {
+		nonZeroMilliCPU, nonZeroMemory := priorityutil.GetNonzeroRequests(&container.Resources.Requests)
+		cpuUsed += nonZeroMilliCPU
+		memoryUsed += nonZeroMemory
+	}
+	for _, container := range pod.Spec.InitContainers {
+		for rName, rQuantity := range container.Resources.Requests {
+			switch rName {
+			case v1.ResourceCPU:
+				if cpu := rQuantity.MilliValue(); cpu > cpuUsed {
+					cpuUsed = cpu
+				}
+			case v1.ResourceMemory:
+				if memory := rQuantity.Value(); memory > memoryUsed {
+					memoryUsed = memory
+				}
+			}
+		}
+	}
+	return cpuUsed, memoryUsed
+}
+
+// makeRoomForCriticalDaemonSetPod ensures that criticalDaemonSet pods have place on the node.
+func makeRoomForCriticalDaemonSetPod(client clientset.Interface, criticalDSPod *v1.Pod, node *v1.Node, policyGroupVersion string, dryRun bool) error {
+	evictablePods, err := podutil.ListEvictablePodsOnNode(client, node)
+	if err != nil {
+		return err
+	}
+	neededCPU, neededMemory := getResourceUsageOfPod(criticalDSPod)
+	var totalMemoryReclaimed, totalCPUReclaimed int64
+	// TODO: @ravig - As of now, there is no order in which pods are getting evicted. We can add logic of
+	// qos here to make this algo better.
+	for _, evictablePod := range evictablePods {
+		if totalMemoryReclaimed >= neededMemory && totalCPUReclaimed >= neededCPU {
+			// We shouldn't enter this in the first iteration.
+			glog.Infof("We don't need to evict any pod futher as critical Pod has enough resources to run on this node")
+			return nil
+		}
+		// As of know, priority is not taken into consideration while evicting.
+		success, err := evictions.EvictPod(client, evictablePod, policyGroupVersion, dryRun)
+		if !success {
```

Comment:
err should be checked for non nil

Reply:
yeah, I just copied that code from existing place. Will modify it. Thanks again.